libminifi/src/FlowControlProtocol.cpp

```diff
@@ -162,11 +164,12 @@ int FlowControlProtocol::sendRegisterReq() {
     return -1;
 
   // Calculate the total payload msg size
-  uint32_t payloadSize = FlowControlMsgIDEncodingLen(FLOW_SERIAL_NUMBER, 0) + FlowControlMsgIDEncodingLen(FLOW_YML_NAME, this->_controller->getName().size() + 1);
-  uint32_t size = sizeof(FlowControlProtocolHeader) + payloadSize;
+  const auto payloadSize = FlowControlMsgIDEncodingLen(FLOW_SERIAL_NUMBER, 0) + FlowControlMsgIDEncodingLen(FLOW_YML_NAME, gsl::narrow<int>(this->_controller->getName().size() + 1));
```

Comment:
I don't see any try catch block around this.

Maybe we should catch and log narrowing exceptions in `FlowControlProtocol::run()`, so that if the thread catches due to them we see the reson.

Reply:
Normally `gsl::narrow<T>` calls `std::terminate()` on failed conversion. I set the config options to change the behavior to throwing an exception in release builds, but we should normally not rely on this.

The motivation here is to fail early at runtime instead of invoking undefined behavior. It should never fail, and if it does, it's a programming error. This should be a hard error, not an exception.